extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2015 Brightcove. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+
+
+class AmpBrightcove extends AMP.BaseElement {
+
+  /** @override */
+  createdCallback() {
+    this.preconnect.url('https://players.brightcove.net');
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    var width = this.element.getAttribute('width');
+    var height = this.element.getAttribute('height');
+    var account = AMP.assert(
+        this.element.getAttribute('data-account'),
+        'The data-account attribute is required for <amp-brightcove> %s',
+        this.element);
+    var playerid = (this.element.getAttribute('data-player-id') || 'default');
+    var embed = (this.element.getAttribute('data-embed') || 'default');
+    var iframe = document.createElement('iframe');
+    var src = 'https://players.brightcove.net/' + account + '/' + playerid + '_' + embed + '/index.html';
+    if (this.element.getAttribute('data-playlist-id')) {
+      src += '?playlistId=' +  this.element.getAttribute('data-playlist-id');
+    } else if (this.element.getAttribute('data-video-id')) {
+      src += '?videoId=' +  this.element.getAttribute('data-video-id');
+    }
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = src;
+    this.applyFillContent(iframe);
+    iframe.width = width;
+    iframe.height = height;
+    this.element.appendChild(iframe);
+    /** @private {?Element} */
+    this.iframe_ = iframe;
+    return loadPromise(iframe);
```

Comment:
Please also whitelist your element in the https://github.com/ampproject/amphtml/blob/master/src/layout.js#L73 - it's a minor thing, but a good touch.